ch_pipeline/processing/daily.py

```diff
@@ -429,10 +513,90 @@ def _available_tags(self):
             csd_set = set(csds)
             csds += [csd for csd in csd_i if csd not in csd_set]
 
+        # grab the list of files that are online, and that exist anywhere, from the earliest csd to the latest
+        csds_sorted = sorted(csds)
+        filenames_online, filenames_that_exist = self._available_files(
+            csds_sorted[0], csds_sorted[-1] + 1
+        )
+
+        # only queue jobs for which all data is available online in chime_online
+        csds_available = self._csds_available_data(
+            csds_sorted, filenames_online, filenames_that_exist
+        )
+        csds = [csd for csd in csds if csd in csds_available]
+
         tags = ["%i" % csd for csd in csds]
 
         return tags
 
+    def _csds_available_data(self, csds, filenames_online, filenames_that_exist):
```

Comment:
This routine will break if `csds` is not sorted. Obviously you feed in a sorted list and so it will be fine in this case, but I think it could be improved if either: the routine did the sort on the csds list itself so it's guaranteed; or at the very least it should make it very clear in the docstring it needs to be sorted.

Reply:
If this routine were to be legitimately used by other functions, it should probably sort everything that it gets. I was hesitent to do that, because everything it gets is sorted already, and for now I would rather include the stipulation in the docstrings.